-------- Original Message --------
Subject: Gaim 0.78 MSN switchboard disconnect fix
Date: Sat, 5 Jun 2004 22:45:26 -0700
From: Christian Hammond <chipx86@gnupdate.org>

This should have gotten into gaim 0.78, but another patch accidentally
reverted it. Basically, sometimes there are harmless errors on MSN
"switchboards," which is where conversations take place. We should
just display the error but keep on going, because unlike errors on
Notification or Nexus servers, these are not fatal. These are usually
things like "The user is offline" and such. However, we were treating
them as fatal, which this patch prevents.

Christian

diff -u -d -p -r1.37 -r1.38
--- gaim/gaim/src/protocols/msn/servconn.c.orig	1 Jun 2004 06:42:20 -0000	1.37
+++ gaim/gaim/src/protocols/msn/servconn.c	2 Jun 2004 04:50:31 -0000	1.38
@@ -58,7 +58,19 @@
 			break;
 	}
 
-	gaim_connection_error(gc, tmp);
+	if (servconn->type != MSN_SERVER_SB)
+		gaim_connection_error(gc, tmp);
+	else
+	{
+		GaimAccount *account = gaim_connection_get_account(gc);
+		char *primary = g_strdup_printf(_("MSN error for account %s"),
+										gaim_account_get_username(account));
+
+		gaim_notify_error(gc, NULL, primary, tmp);
+
+		g_free(primary);
+	}
+
 
 	g_free(tmp);
 }
